torch/distributed/_shard/sharded_tensor/metadata.py

```diff
@@ -58,6 +58,15 @@ def __setstate__(
 
         self.memory_format = memory_format
 
+    @staticmethod
+    def create_for_tensor(tensor: torch.Tensor) -> "TensorProperties":
```

Comment:
I am wondering if this function should be renamed to reflect its meaning?

Reply:
Yeah, a better name that follows the rest of PTD convention is `create_from`, I don't think we need to use `create_properties_from_tensor` since this is a class method so `TensorProperties.create_from_tensor` should be good.